### PR TITLE
Add Map.filterMap

### PR DESCRIPTION
**Pull request by pm5**
Moved over from darklang/tablecloth#273.

### CODE REVIEW

File: native/src/TableclothMap.ml [hunk 1]
```diff
@@ -86,6 +86,12 @@ let map_with_index t ~f = Base.Map.mapi t ~f:(fun ~key ~data -> f key data)
 
 let filter = Base.Map.filter
 
+let filterMap m ~f =
```
Comment (pbiggar) [inline native/src/TableclothMap.ml]: We no longer use camelCase for OCaml

File: rescript/src/TableclothMap.ml [hunk 2]
```diff
@@ -36,6 +36,17 @@ let mapWithIndex t ~f = Belt.Map.mapWithKey t f
 
 let filter m ~f = Belt.Map.keep m (fun _ value -> f value)
 
+let filterMap m ~f =
```
Comment (pbiggar) [inline rescript/src/TableclothMap.ml]: Rescript is no longer in this repo (it's in https://github.com/darklang/tablecloth-rescript)

Comment (pbiggar) [reply rescript/src/TableclothMap.ml]: Would love a PR there.

File: rescript/test/MapTest.ml [hunk 3]
```diff
@@ -75,6 +75,33 @@ let suite =
           test "has length zero" (fun () ->
               expect (Tablecloth.Map.empty (module Char) |> Map.length)
               |> toEqual Eq.int 0 ) ) ;
+
+      describe "filterMap" (fun () ->
```
Comment (pbiggar) [inline rescript/test/MapTest.ml]: This test should be for native.

**Review by pbiggar — CHANGES_REQUESTED** [review]
Thanks! Made a few notes - might be better to wait until after #6

**Comment by pm5** [comment]
If we're done with #9 then I think this is ready.  I'll move the Rescript part to https://github.com/darklang/tablecloth-rescript

**Comment by pbiggar** [comment]
Thanks!

**Pull request merged**